packages/design-system/ui/src/hooks/use-prompt/use-prompt.tsx

```diff
@@ -1,22 +1,33 @@
 "use client"
 
 import * as React from "react"
+import { useRef } from "react"
 import { createRoot } from "react-dom/client"
 import { RenderPrompt, RenderPromptProps } from "./render-prompt"
 
 type UsePromptProps = Omit<RenderPromptProps, "onConfirm" | "onCancel" | "open">
 
 const usePrompt = () => {
+  const isPromptActive = useRef(false)
+  const currentPromptPromise = useRef<Promise<boolean> | null>(null)
+
   const prompt = async (props: UsePromptProps): Promise<boolean> => {
-    return new Promise((resolve) => {
-      let open = true
+    if (isPromptActive.current) {
+      // If the prompt is already active, return the current promise
+      return currentPromptPromise.current!
```

Comment:
Instead of returning `false` we return the active promise if `prompt` is called again while the prompt is already active.